bin/strata-client/src/rpc_server.rs

```diff
@@ -298,38 +298,24 @@ impl<D: Database + Send + Sync + 'static> StrataApiServer for StrataRpcImpl<D> {
         }
     }
 
-    async fn get_cl_block_witness_raw(&self, idx: u64) -> RpcResult<Option<Vec<u8>>> {
-        let blk_manifest_db = self.database.clone();
-        let blk_ids: Vec<L2BlockId> = wait_blocking("l2_blockid", move || {
-            blk_manifest_db
-                .clone()
-                .l2_db()
-                .get_blocks_at_height(idx)
-                .map_err(Error::Db)
-        })
-        .await?;
-
-        // Check if blk_ids is empty
-        let blkid = match blk_ids.first() {
-            Some(id) => id.to_owned(),
-            None => return Ok(None),
-        };
-
+    async fn get_cl_block_witness_raw(&self, blkid: L2BlockId) -> RpcResult<Option<Vec<u8>>> {
```

Comment:
I think this should not return an option. It might make sense to return Error when the information is not found.